front/components/assistant_builder/ActionsScreen.tsx

```diff
@@ -0,0 +1,452 @@
+  AssistantBuilderActionConfiguration["type"],
+  string
+> = {
+  RETRIEVAL_SEARCH: "Search",
+  RETRIEVAL_EXHAUSTIVE: "Exhaustive Search",
+  PROCESS: "Process",
+  DUST_APP_RUN: "Dust App",
+  TABLES_QUERY: "Tables Query",
+};
+
+function ActionModeSection({
+  children,
+  show,
+}: {
+  children: ReactNode;
+  show: boolean;
+}) {
+  return show && <div className="flex flex-col gap-6">{children}</div>;
+}
+
+export function isActionValid(
+  action: AssistantBuilderActionConfiguration
+): boolean {
+  switch (action.type) {
+    case "RETRIEVAL_SEARCH":
+      return isActionRetrievalSearchValid(action);
+    case "RETRIEVAL_EXHAUSTIVE":
+      return isActionRetrievalExhaustiveValid(action);
+    case "PROCESS":
+      return isActionProcessValid(action);
+    case "DUST_APP_RUN":
+      return isActionDustAppRunValid(action);
+    case "TABLES_QUERY":
+      return isActionTablesQueryValid(action);
+    default:
+      assertNever(action);
+  }
+}
+
+export default function ActionsScreen({
+  owner,
+  builderState,
+  dustApps,
+  dataSources,
+  setBuilderState,
+  setEdited,
+}: {
+  owner: WorkspaceType;
+  builderState: AssistantBuilderState;
+  dataSources: DataSourceType[];
+  dustApps: AppType[];
+  setBuilderState: (
+    stateFn: (state: AssistantBuilderState) => AssistantBuilderState
+  ) => void;
+  setEdited: (edited: boolean) => void;
+}) {
+  const [newActionModalOpen, setNewActionModalOpen] = React.useState(false);
+  const [actionToEdit, setActionToEdit] =
+    React.useState<AssistantBuilderActionConfiguration | null>(null);
+
+  const upsertAction = useCallback(
+    (newAction: AssistantBuilderActionConfiguration) => {
+      setEdited(true);
+      setBuilderState((state) => {
+        let found = false;
+        const newActions = state.actions.map((action) => {
+          if (action.name === newAction.name) {
+            found = true;
+            return newAction;
+          }
+          return action;
+        });
+        if (!found) {
+          newActions.push(newAction);
+        }
+        return {
+          ...state,
+          actions: newActions,
+        };
+      });
+    },
+    [setBuilderState, setEdited]
+  );
+
+  const updateAction = useCallback(
+    (name: string, newAction: AssistantBuilderActionConfiguration) => {
+      setEdited(true);
+      setBuilderState((state) => {
+        return {
+          ...state,
+          actions: state.actions.map((a) => (a.name === name ? newAction : a)),
+        };
+      });
+    },
+    [setBuilderState, setEdited]
+  );
+
+  const deleteAction = useCallback(
+    (name: string) => {
+      setEdited(true);
+      setBuilderState((state) => {
+        return {
+          ...state,
+          actions: state.actions.filter((a) => a.name !== name),
+        };
+      });
+    },
+    [setBuilderState, setEdited]
+  );
+
+  const allActionsValid = builderState.actions.every(isActionValid);
+
+  return (
+    <>
+      <NewActionModal
```

Comment:
If we could addd something to next to each action maybe? having a modal for just name and description seems horribly unfriendly to use yet the alternative (even if dirty) is easy to implement and easier to use?

Reply:
name desciption and type :)

This whole thing will change anyway, let's just get this out (it's very simple with the modal) and make it nice once we have a more elaborate design ?

Reply:
I really don't understand the need for a modal still but no problem shipping this is as is